### ISSUE TITLE
SQL snapshots index on managed type

### ISSUE THREAD

**Issue by tilgovi** [javers/javers#1288]
Executing a `findSnapshots` query like the following results in a full table scan:

```java
    javers.findSnapshots(QueryBuilder.byClass(MyClass.class).limit(30).build())
```

It might be nice if this query could use an index on the `managed_type` column to avoid scanning rows with snapshots for other types.

I'm not sure what alternative exists.

This would be useful for finding snapshots of entities that change infrequently when other entities change much more frequently. As it is now, it may be necessary to scan many unrelated snapshot rows to find ones for the given type.

**Comment by bartoszwalacik** [comment]
okay, it's about mongo or SQL repository? I'm sure there are some indexes on the Snapshot table/collection. could you please post the list of existing indexes here?

**Comment by tilgovi** [comment]
SQL, as specified in the issue title.

Default schema factory creates indexes on the foreign key, but not the managed type.

https://github.com/javers/javers/blob/master/javers-persistence-sql/src/main/java/org/javers/repository/sql/schema/FixedSchemaFactory.java#L65

**Comment by tilgovi** [comment]
Snapshot queries are ordered by descending primary key, so fetching snapshots is perfectly fast unless the entity type occurs infrequently. If there are many snapshots, and relatively few snapshots for some type, querying for the uncommon type scans many rows for snapshots of other types.

It's okay if this is intentional, and it turns out I didn't have a need for this, but I noticed this difference between MongoDB and SQL implementations and thought I should raise the discussion.

**Comment by bartoszwalacik** [comment]
You are right, indexes should be similar for both repositories, would you like to contribute this index addition?

**Comment by bartoszwalacik** [comment]
@tilgovi could you please give a concrete query which results in a full scan? Because indexes related to types are present in the GLOBAL_ID table, see https://github.com/javers/javers/blob/b83fbd1176c3ea10d7ecfc167b948a14959ccd8a/javers-persistence-sql/src/main/java/org/javers/repository/sql/schema/FixedSchemaFactory.java#L138C76-L138C76
